src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -1066,14 +1077,17 @@ void SolverInterfaceImpl::writeScalarData(
   PRECICE_TRACE(fromDataID, valueIndex, value);
   PRECICE_CHECK(_state != State::Finalized, "writeScalarData(...) cannot be called after finalize().");
   PRECICE_VALIDATE_DATA_ID(fromDataID);
-  PRECICE_CHECK(valueIndex >= -1, "Invalid value index (" << valueIndex << ") when writing scalar data!");
+  PRECICE_CHECK(valueIndex >= -1, "Invalid value index (" << valueIndex << ") when writing scalar data. Value index must be >= 0.");
```

Comment:
Why changing the phrasing here compared to above? It would also be nice to have the context here, similarly to:

```c++
PRECICE_CHECK(0 <= valueIndex && valueIndex < valuesInternal.size() / context.fromData->getDimensions(), "Value index out of range. Please check that the size of "
<< context.fromData->getName() << " is correct.");
```

Reply:
The same in two more instances below.